examples/architecture/JSONCtrProperties.wyt

```diff
@@ -0,0 +1,81 @@
+import wyvern.collections.list
+import wyvern.archast
+import wyvern.ast
+import wyvern.option
+
+type JSONCtrProperties
+	val host : String
+	val prt : Int
+	metadata new
+		def checkPortCompatibility(l : list.List[archast.PortDecl]) : Int
+			val length = l.length()
+			type PortDecl = archast.PortDecl
+			var p1 : PortDecl = archast.NonePort
+			var p2 : PortDecl = archast.NonePort
+			if (l.length() == 2)	
+					val op1 : option.Option[PortDecl] = l.get(0)
+					val op2 : option.Option[PortDecl] = l.get(1)
+					if (op1.isDefined && op2.isDefined)
+							p1 = op1.getOrElse(() => p1)
+							p1 = op2.getOrElse(() => p1)
+						else 
+							-1
+					var requires : String = ""
+					var provides : String = ""
+					if (p1.getRequires() == "")
+							if (p1.getProvides() == "")
+									-1
+								else
+									provides = p1.getProvides()
+						else
+							requires = p1.getRequires()
+					if (requires == "")	
+							requires = p2.getRequires()
+						else	
+							provides = p2.getProvides()
+								
+					if (requires == provides)
+							length
+						else
+							-1
+					length
+				else
+					-1
+
+		def generateConnectorImpl(l : list.List[archast.PortDecl]) : list.List[ast.AST]
+			if (this.checkPortCompatibility(l) > 0)
+					var connectorIface : String = ""
+					val portDecl = l.get(0).getOrElse(() => archast.NonePort)
```

Comment:
Since this is done after a port compatibility check, you can eliminate the use of NonePort here by doing the following:

val portDecl = l.get(0).value

Reply:
I get an error when trying to access the value field like above.I  think since type Option does store a val value in it and it's only added when Some() is called to create a new Option, it's not accessible.

Reply:
Ah, I see! I guess it is okay to have archast.NonePort then.